### PR TITLE
refactored

### PR DESCRIPTION
**Pull request by t-keazirian**
* removed Console Write Lines
* added dispose and closing socket calls

### CODE REVIEW

File: TKeazirian.HTTPServer/EchoServer.cs [hunk 1]
```diff
@@ -19,16 +19,18 @@ public static void StartListening()
...
 
                 _request = GetRequest(handler);
 
                 var responseToSend = Controller.GenerateResponse(_request);
-                handler.Send(responseToSend);
+
+                handler.NoDelay = true;
+
+                handler.Send(responseToSend, SocketFlags.None);
+                handler.Close();
```
Comment (Aoxian) [inline TKeazirian.HTTPServer/EchoServer.cs:33]: Is `handler` a good variable name? A handler is something that typically runs in response to an event.  What is the triggering event here?

File: TKeazirian.HTTPServer/SocketHandler.cs [hunk 2]
```diff
@@ -15,5 +15,6 @@ public static void CloseSocketConnection(Socket handler)
     {
         handler.Shutdown(SocketShutdown.Both);
         handler.Close();
+        handler.Dispose();
```
Comment (Aoxian) [inline TKeazirian.HTTPServer/SocketHandler.cs:18]: > added dispose and closing socket calls

Is a `handler` actually a socket? I think I am confused here.

File: TKeazirian.HTTPServer/EchoServer.cs [hunk 3]
```diff
@@ -52,7 +51,6 @@ public static string GetRequest(Socket handler)
         byte[] bytes = new byte[1024];
         int bytesReceived = handler.Receive(bytes);
         _request = Encoding.ASCII.GetString(bytes, 0, bytesReceived);
-        Console.WriteLine($"Text received: {_request}");
```
[the remark this answers is not in the document]
Comment (t-keazirian) [reply TKeazirian.HTTPServer/EchoServer.cs:55]: Yes, that makes sense.  Marie and I had talked about how if I'm using Azure, Azure does not have a console.  But you are correct, for the purposes of this project right now, I should/can add them back in.

File: TKeazirian.HTTPServer/SocketHandler.cs (hunk 2 above)
Comment (t-keazirian) [reply TKeazirian.HTTPServer/SocketHandler.cs:18]: Yes, the `handler` is the socket.  I can call it `socket` to be more clear.  It is originally `listener` when it is first instantiated (in `CreateSocketListener`) and then I rename it `handler`.  But `socket` will be more clear, I think?

File: TKeazirian.HTTPServer/Controller.cs [hunk 4]
```diff
@@ -15,5 +15,13 @@ public static byte[] GenerateResponse(string request)
...
             return responseToSend;
         }
+
+        public static byte[] GenerateErrorResponse()
+        {
+            var errorResponse = "HTTP/1.1 500 Internal Server Error";
+
+            var errorResponseToSend = Encoding.ASCII.GetBytes(errorResponse);
+            return errorResponseToSend;
+        }
```
Comment (Aoxian) [inline TKeazirian.HTTPServer/Controller.cs:25]: The `Controller` seems to do several things.  It builds an "Echo body" response it builds an `ErrorResponse`, and it also appears to be doing all your parsing.  What is the problem you are trying to solve and is there a way to break this up further?

Comment (Aoxian) [inline TKeazirian.HTTPServer/Controller.cs:25]: The `Controller` seems to do several things.  It builds an "Echo body" response it builds an `ErrorResponse`, and it also appears to be doing all your parsing.  What is the problem you are trying to solve and is there a way to break this up further?

Comment (Aoxian) [inline TKeazirian.HTTPServer/Controller.cs:25]: The `Controller` seems to do several things.  It builds an "Echo body" response it builds an `ErrorResponse`, and it also appears to be doing all your parsing.  What is the problem you are trying to solve and is there a way to break this up further?

File: TKeazirian.HTTPServer/SocketHandler.cs (hunk 2 above)
Comment (Aoxian) [reply TKeazirian.HTTPServer/SocketHandler.cs:18]: If its actually a socket, I'd call it a socket.  It HTTP definitions, a `handler` would handle an event.

**Review by MarieRobinson — APPROVED** [review]
nice refactoring.

**Review by Aoxian — APPROVED** [review]
This is fine for now. Would like to know what your architectural plans are, but that is outside of this PR as you appear.

**Pull request merged**